### PR TITLE
network-ng: s390 autoyast configuration

### PR DESCRIPTION
**Pull request by teclator**
- https://trello.com/c/D3n3PbVH/1278-5-network-ng-autoyast-support

### CODE REVIEW

File: src/lib/network/network_autoyast.rb [hunk 1]
```diff
@@ -113,6 +117,28 @@ def configure_lan(write: false)
...
+    def activate_s390_devices(section = nil)
+      profile_devices = section || ay_networking_section["s390-devices"] || {}
+      devices_section = Y2Network::AutoinstProfile::S390DevicesSection.new_from_hashes(profile_devices)
+      connections = Y2Network::Autoinst::S390DevicesReader.new(devices_section).config
+      connections.each do |conn|
+        begin
+          builder = Y2Network::InterfaceConfigBuilder.for(conn.type, config: conn)
+          activator = Y2Network::S390DeviceActivator.for(builder)
+          if activator.configure
+            builder.name = activator.configured_interface
```
Comment (jreidinger) [inline src/lib/network/network_autoyast.rb]: what is the goal is assigning name to builder when you do not save it?

File: src/lib/y2network/autoinst_profile/s390_device_section.rb [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+      # @param config [Y2Network::ConnectionConfig]
+      # @return [Boolean]
+      def init_from_config(config)
+        @type = config.type.short_name
+        case config
+        when ConnectionConfig::Qeth
+          @chanids = config.device_id
+          @layer2 = config.layer2
+        when ConnectionConfig::Ctc
+          @protocol = config.protocol
```
Comment (jreidinger) [inline src/lib/y2network/autoinst_profile/s390_device_section.rb]: what about LCS?

File: src/modules/LanItems.rb [hunk 3]
```diff
@@ -401,7 +402,7 @@ def Import(settings)
       # - creates s390 device eth emulation
       # So, it belongs partly into Import and partly into Write. Note, that
       # the code is currently unable to revert already created emulated device.
-      settings.fetch("s390-devices", {}).each { |rule| createS390Device(rule) } if Arch.s390
+      NetworkAutoYast.instance.activate_s390_devices(settings.fetch("s390-devices", {}))
```
Comment (jreidinger) [inline src/modules/LanItems.rb]: I would keep here that s390 condition, so it do not try fancy things if profile contain it accidentally.

**Comment by jreidinger** [comment]
just few notes. overall looks good.

File: src/lib/network/network_autoyast.rb (hunk 1 above)
Comment (teclator) [reply src/lib/network/network_autoyast.rb]: It is not needed at all. It makes sense when activating the device for editing, but in AY it is irrelevant.

File: package/yast2-network.changes [hunk 4]
```diff
@@ -1,3 +1,9 @@
+-------------------------------------------------------------------
+Mon Oct  7 06:05:22 UTC 2019 - Knut Anderssen <kanderssen@suse.com>
+
+- AutoYaST: Added back the activation of s390 devices during the
```
Comment (jreidinger) [inline package/yast2-network.changes]: maybe add added back after what. Like `after switch to new data model`. So it is clear it is not something broken in SP1.

**Pull request merged**